internal/services/issuer/verifiable_credential.go

```diff
@@ -180,3 +189,82 @@ func (i *Issuer) VIN(vin string, tokenID *big.Int) (id string, err error) {
 
 	return id, tx.Commit()
 }
+
+func (i *Issuer) Handle(ctx context.Context, event *ADVinCredentialEvent) error {
+	observedVIN, err := services.ExtractVIN(event.CloudEvent.Data)
+	if err != nil {
+		i.log.Info().Err(err).Msg("could not extract vin from payload")
+		return err
+	}
+
+	logger := i.log.With().Str("device-address", event.Subject).Str("vin", observedVIN).Logger()
+	logger.Info().Msg("got vin credentialer event")
+
+	ad, err := models.AftermarketDevices(
+		models.AftermarketDeviceWhere.EthereumAddress.EQ(
+			null.BytesFrom(common.FromHex(event.Subject)),
+		),
+		qm.Load(models.AftermarketDeviceRels.VehicleToken),
+	).One(ctx, i.DBS.DBS().Reader)
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			i.log.Info().Err(err).Msg("no corresponding aftermarket device for address")
+			return err
+		}
+		i.log.Info().Err(err).Msg("database failure retrieving aftermarket device")
+		return err
+	}
+
+	vnft, err := models.VehicleNFTS(
+		models.VehicleNFTWhere.TokenID.EQ(ad.VehicleTokenID),
+		qm.Load(models.VehicleNFTRels.UserDevice),
+		qm.Load(models.VehicleNFTRels.Claim, qm.OrderBy(models.VerifiableCredentialColumns.ExpirationDate+" DESC")),
+	).One(ctx, i.DBS.DBS().Reader)
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			logger.Err(err).Msg("no associated vehicle nft found for device")
+			return err
+		}
+		i.log.Info().Err(err).Msg("database failure retrieving nft associated with device")
+		return err
+	}
+
+	if !vnft.R.UserDevice.VinConfirmed {
+		err := errors.New("invalid vin")
+		logger.Err(err).Msg("vin associated with device not confirmed")
+		return err
+	}
+
+	if vnft.R.UserDevice.VinIdentifier.String != observedVIN {
+		// do we want to do anything here?
+		logger.Info().Msg("observed vin does not match confirmed vin")
+		return nil
+	}
+
+	if vnft.R.Claim != nil {
+		if vnft.R.Claim.ExpirationDate.After(time.Now()) {
```

Comment:
I think we'd like to roughly make sure that the user is good for the next drop.

Maybe we check that the expiration date is after the end time of the current rewards week? There's some boundary-finding code in rewards-api.